### PR TITLE
feat(elevation_map_loader): use polygon iterator to speed up

### PR DESCRIPTION
**Pull request by miursh**
## Description

When use_lane_filter is true, elevation map loader makes inpaint mask only within the lane polygons.
This option could be useful for large size map which takes long time to inpaint.

## Related links

[TIERIV COMPANY INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-25851)

## Tests performed

We have confirmed below
* The elevation value is completely same within lanelet if margin is larger than 1m
* Calculation time is  much shorter than inpainting the entire area. (About 4 hours to 3 mins)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/elevation_map_loader/src/elevation_map_loader_node.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include "elevation_map_loader/elevation_map_loader_node.hpp"
 
 #include <grid_map_core/GridMap.hpp>
+#include <grid_map_core/iterators/PolygonIterator.hpp>
```
Comment (maxime-clem) [inline perception/elevation_map_loader/src/elevation_map_loader_node.cpp:18]:
```suggestion
#include <grid_map_utils/PolygonIterator.hpp>
```

File: perception/elevation_map_loader/src/elevation_map_loader_node.cpp [hunk 2]
```diff
@@ -215,9 +216,25 @@ void ElevationMapLoaderNode::inpaintElevationMap(const float radius)
...
-    if (!elevation_map_.isValid(*iterator, layer_name_)) {
-      elevation_map_.at("inpaint_mask", *iterator) = 1.0;
+  if (lane_filter_.use_lane_filter_) {
+    for (const auto & lanelet : lane_filter_.road_lanelets_) {
+      auto lane_polygon = lanelet.polygon2d().basicPolygon();
+      grid_map::Polygon polygon;
+      for (const auto & p : lane_polygon) {
+        polygon.addVertex(grid_map::Position(p[0], p[1]));
+      }
+      for (grid_map::PolygonIterator iterator(elevation_map_, polygon); !iterator.isPastEnd();
```
Comment (maxime-clem) [inline perception/elevation_map_loader/src/elevation_map_loader_node.cpp:226]:
```suggestion
      for (grid_map_utils::PolygonIterator iterator(elevation_map_, polygon); !iterator.isPastEnd();
```

**Review by maxime-clem — COMMENTED** [review]
For better performances I recommend using the `grid_map_utils::PolygonIterator` (https://github.com/autowarefoundation/autoware.universe/tree/main/common/grid_map_utils).
You only need to add `grid_map_utils` to `packages.xml` and apply the suggested changes.

File: launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py [hunk 3]
```diff
@@ -324,7 +324,7 @@ def create_single_frame_outlier_filter_components(input_topic, output_topic, con
                 ],
                 parameters=[
                     {
-                        "use_lane_filter": False,
+                        "use_lane_filter": True,
```
Comment (yukkysaito) [inline launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py:327]: This is debatable. The detection area is looking at pointcloud, but it is often defined outside of ll2 lanes.

Comment (Shin-kyoto) [reply launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py:327]: Even if we use lane filter for inpaint, the elevation value in cells including `min_num_points_per_cell` or more points outside of lanelet is calculated. 
This PR influences only cells outside of lanelet that do not include `min_num_points_per_cell` points.

(And if we can expand Polygon of lanelet, we can include cells in detection_area outside of lanelet.)

Comment (yukkysaito) [reply launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py:327]: I see. :+1:  I was mistaken.

Comment (miursh) [reply launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py:327]: @yukkysaito Although, setting default to True is not good idea, I think. I will fix this.
And this is a bit confusing, is it better to move these parameters to ground_segmentation.param.yaml? What do you think?

File: perception/elevation_map_loader/src/elevation_map_loader_node.cpp (hunk 2 above)
Comment (miursh) [reply perception/elevation_map_loader/src/elevation_map_loader_node.cpp]: @maxime-clem Thanks for your advice!
fixed https://github.com/autowarefoundation/autoware.universe/pull/2885/commits/666caa1403396967ffe605f822a27e100510ae9c

**Review by Shin-kyoto — APPROVED** [review]
LGTM

**Comment by miursh** [comment]
@kosuke55 @taichiH Could you review this PR as codeowners?

@Shin-kyoto and I have already checked it works all right

**Review by kosuke55 — APPROVED** [review]
LGTM

**Review by taichiH — APPROVED** [review]

**Pull request merged**